Parse JSON Section in Sample Code (#2873)
nasa-gcn/gcn.nasa.gov: app/routes/docs.client.samples.md

```python
import base64

# Convert the Kafka message value to a string
value_str = value.decode("utf-8")

alert_json = json.loads(value_str)

skymap_string = alert_json["event"]["skymap"]

# Decode the Base64 string
decoded_bytes = base64.b64decode(skymap_string)

with open("skymap.fits", "wb") as fits_file:
    fits_file.write(decoded_bytes)
```

If you want to include a FITS file in a Notice, you add a property to your schema definition in the following format:

```python
{
    type: 'string',
	contentEncoding: 'base64',
	contentMediaType: 'image/fits',
}
```

## Encoding Embedded Data

For producer data production pipelines, the following encoding steps convert an input file to a bytestring. This guide demonstrates how to encode a file (e.g., skymap.fits) into a `base64` encoded string and submit it to the GCN Kafka broker.

```python
from gcn_kafka import Producer
import base64

# Set Kafka Topic and Producer Configuration
TOPIC = "igwn.gwalert"

> producer = Producer(client_id='fill me in', # Replace with your client ID

Author: Q: Why do you need this?

Re: As fits file will be encoded and sent via Kafka from the producer's side.

Reviewer: What does this refer to? It sounds like you are referring to a comment from an earlier PR. Please link to that comment.

Author: PR #2669 
<img width="1057" alt="Screenshot 2025-02-12 at 1 28 42 AM" src="https://github.com/user-attachments/assets/ef33e447-44c2-4424-9c29-98d03efd0c66" />

Reviewer 2: https://github.com/nasa-gcn/gcn.nasa.gov/pull/2669#discussion_r1939641730

    print(message.value())
```

## Parsing JSON

GCN Notices for new missions are typically distributed in JSON format. This guide explains how to programmatically read the JSON schema.

Start with subscribing to a Kafka topic and parsing the JSON data

```python
from gcn_kafka import Consumer
import json

# Connect as a Kafka consumer

> consumer = Consumer(

Author: Q: What's this for?

Re: Shall we remove setting-up consumer and producer connect on Kafka?

Author: https://github.com/nasa-gcn/gcn.nasa.gov/pull/2669#discussion_r1952060456

    client_id='fill me in', # Replace with your client ID
    client_secret='fill me in', # Replace with your client secret
    config={"message.max.bytes": 204194304},
)

# Subscribe to Kafka topic
consumer.subscribe(['igwn.gwalert'])

# Continuously consume and parse JSON data
for message in consumer.consume(timeout=1):
    if message.error():
        print(message.error())
        continue

    # Print the topic and message ID
    print(f"topic={message.topic()}, offset={message.offset()}")

>     # Kafka message value as a Base64-encoded string

Author: ```suggestion
    # Kafka message value as a JSON dictionary encoded as a string or byte array
```

    print(message.value())
```

## Parsing JSON

GCN Notices are distributed through Kafka topics and, for new missions, are typically provided in JSON format. This guide explains how to programmatically read the JSON schema.

Start by subscribing to a Kafka topic and parsing the JSON data.

```python
from gcn_kafka import Consumer
import json

# Kafka consumer configuration
consumer = Consumer(
    client_id='fill me in',
    client_secret='fill me in',
    config={"message.max.bytes": 204194304},
)

# Subscribe to Kafka topic
consumer.subscribe(['igwn.gwalert'])

# Continuously consume and parse JSON data
for message in consumer.consume(timeout=1):
    if message.error():
        print(message.error())
        continue

    # Print the topic and message ID
    print(f"topic={message.topic()}, offset={message.offset()}")

    # Decode message value
    value = message.value()
    value_str = value.decode("utf-8")
    alert_json = json.loads(value_str)
```

## Decoding Embedded Data

> Some GCN notices include HEALPix sky maps encoded in `base64`.

Reviewer: ```suggestion
Some GCN notices include HEALPix sky maps encoded in base64.
```

Also provide a link to some reference for what base64 is.

Reviewer: Thanks for applying this suggestion, but please also remember to turn the text "base64" into a link to some page that explains what it is.

Author: Done, was working on it.

The following code demonstrates how to extract, decode, and save the HEALPix data as a `.fits` file from a received notice. Python's built-in [`base64`](https://docs.python.org/3/library/base64.html#base64.b64encode) module provides the `b64decode` method to simplify the decoding process.

```python
import base64
from astropy.io import fits

# Extract the base64-encoded skymap
skymap_string = alert_json["event"]["skymap"]

# Decode the Base64 string to bytes
decoded_bytes = base64.b64decode(skymap_string)

# Write bytes to a FITS file
with open("skymap.fits", "wb") as fits_file:
    fits_file.write(decoded_bytes)

# Open and inspect the FITS file
with fits.open("skymap.fits") as hdul:
    hdul.info()
```

## JSON Schema Example for Embedding a FITS File

If you want to include a FITS file in a Notice, add a property to your schema definition in the following format:

```json
"healpix_file": {
    "type": "string",
    "contentEncoding": "base64",
    "contentMediaType": "image/fits",
    "description": "Base 64 encoded content of a FITS file"
}
```

> JSON Schema supports embedding non-JSON media within strings using the `contentMediaType` and `contentEncoding`, which enable the distribution of diverse data types. For further details, refer to [non-JSON data](https://json-schema.org/understanding-json-schema/reference/non_json_data.html).

Reviewer: ```suggestion
JSON Schema supports embedding non-Unicode media within strings using the `contentMediaType` and `contentEncoding`, which enable the distribution of diverse data types. For further details, refer to [non-JSON data](https://json-schema.org/understanding-json-schema/reference/non_json_data.html).
```
